modules/commerce_bluesnap_currency/commerce_bluesnap_currency.module

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/**
+ * @file
+ * Provides Bluesnap currency resolver settings.
+ */
+
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Implements hook_form_alter().
+ */
+function commerce_bluesnap_currency_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  // Alter commerce_currency_resolver_admin_conversion form and add
+  // bluesnap api settings.
+  if ($form_id == "commerce_currency_resolver_admin_conversion") {
+    // Get current settings.
+    $config = \Drupal::config('commerce_currency_resolver.currency_conversion');
+
+    $form['bluesnap'] = [
+      '#type' => 'details',
+      '#title' => t('Bluesnap Settings'),
+      '#open' => FALSE,
+      '#tree' => TRUE,
+    ];
```

Comment:
**CHANGE REQUEST**

Show/hide using AJAX states only if BlueSnap is selected as the exchange rate provider.

Reply:
Done